lib/sentry/opentelemetry/span_storage.ex

```diff
@@ -0,0 +1,151 @@
+defmodule Sentry.OpenTelemetry.SpanStorage do
+  @moduledoc false
+  use GenServer
+
+  defstruct [:cleanup_interval, :table_name]
+
+  alias Sentry.OpenTelemetry.SpanRecord
+
+  @cleanup_interval :timer.minutes(5)
+  @span_ttl :timer.minutes(30)
+
+  @spec start_link(keyword()) :: GenServer.on_start()
+  def start_link(opts) when is_list(opts) do
+    name = Keyword.get(opts, :name, __MODULE__)
+    GenServer.start_link(__MODULE__, opts, name: name)
+  end
+
+  @impl true
+  def init(opts) do
+    table_name = Keyword.get(opts, :table_name, default_table_name())
```

Comment:
If the name of this server is customizable, I don't think we need to make this customizable too. We can just always do `Module.concat(name, ETSTable)`.